docs(kilo-docs): document MCP tool permission levels and cross-link from related pages (#8546)
Kilo-Org/kilocode: packages/kilo-docs/pages/automate/mcp/using-in-kilo-code.md

{% /tab %}
{% /tabs %}

### Auto Approve Tools
### Tool Permissions

{% tabs %}
{% tab label="VSCode" %}

MCP tool calls use the same permission system as built-in tools. Each MCP tool's permission key is its namespaced name: `{server}_{tool}` (e.g. `my_server_do_something`).
MCP tools use the same permission system as built-in tools (`allow`, `ask`, `deny`). Each MCP tool's permission key is its namespaced name: `{server}_{tool}` (e.g. `my_server_do_something`). You can use glob patterns like `my_server_*` for broad rules.

> **At runtime:** When an MCP tool is called, the Permission Dock shows an approval prompt. Click **Approve Always** to save an allow rule to your config so future calls to that tool are auto-approved.

Author: i like this wording beter actually, lets keep it.

**In your config file:** Add the tool name (or a wildcard pattern) to the `permission` key in `kilo.jsonc`:

> ```json

Author: example inline is good

{
  "permission": {

>     "my_server_do_something": "allow",

Author: Default is allow, change these to for example the GitHub MCP tool for  creating a PR or another write-operation that folks may want to specifically grant permission to.

> For full details on configuring MCP tool permissions — including examples with glob patterns and per-tool overrides — see [Auto-Approving Actions](/docs/getting-started/settings/auto-approving-actions#mcp-tool-permissions).

Author: ```suggestion
For full details on configuring auto-approval permissions — see [Auto-Approving Actions](/docs/getting-started/settings/auto-approving-actions).
```

> {% tabs %}

Author: is this correct? we don't want a new tab section

> In the legacy extension, MCP tool auto-approval works on a per-tool basis and is disabled by default. To configure auto-approval:

Author: leave this section alone
